internal/core/http.go

```diff
@@ -129,6 +131,10 @@ func (r HTTPRoute) GetType() ResolvedRouteType {
 	return ResolvedHTTPRouteType
 }
 
+func (r HTTPRoute) Marshal() ([]byte, error) {
+	return json.Marshal(r)
```

Comment:
Is there a reason we need these instead of just calling json.Marshal(r) directly? Just convenience?